src/crystal/system/unix/time.cr

```diff
@@ -0,0 +1,37 @@
+require "c/sys/time"
+require "c/time"
+
+module Crystal::System::Time
+  def self.compute_offset(second)
+    LibC.tzset
+    offset = nil
+
+    {% if LibC.methods.includes?("daylight".id) %}
+      if LibC.daylight == 0
+        # current TZ doesn't have any DST, neither in past, present or future
+        offset = -LibC.timezone.to_i64
+      end
+    {% end %}
+
+    unless offset
+      # current TZ may have DST, either in past, present or future
+      ret = LibC.localtime_r(pointerof(second), out tm)
+      raise Errno.new("localtime_r") if ret.null?
+      offset = tm.tm_gmtoff.to_i64
+    end
+
+    offset / 60 * TicksPerMinute
+  end
+
+  def self.compute_second_and_tenth_microsecond
+    {% if flag?(:darwin) %}
```

Comment:
I'd move it to a separate file. I think inside a given System implementation there shouldn't be any `if flag?`, otherwise it kind of missing the point of extracting it into a different place.